docs/pages/en-us/Error-Handling.md

```diff
@@ -0,0 +1,114 @@
+2. The processing of the message causes the worker to die unexpectedly.
+3. The Autoscaler service scales the worker back up again to facilitate a retry.
+4. The original message is re-delivered for retry where the `x-caf-worker-retry` message header is stamped on the message and incremented. Messages which cause a worker to crash are retried up to 10 times by default before the worker gives up trying to process them and marks them as poisoned. Note that the number of permitted retries is configurable within the `RabbitWorkerQueueConfiguration` file for a worker.
+5. Once the permitted number of retries is exceeded, the original message will be acknowledged.
+6. A `RESULT_EXCEPTION` response including the poisoned message details is placed on the messaging output queue `worker-output-queue` by the worker framework when the permitted number of retries is exceeded.
+
+## OutOfMemory / StackOverflow Errors
+Messages which cause OutOfMemory / StackOverflow errors also cause the Worker to crash. These are treated the same as poisoned messages and will be retried until successful or until the retry count exceeds the permitted number of retries.  If the permitted number of retries is exceeded the message will be placed on the `worker-output-queue` by the framework, with a task status of “RESULT_EXCEPTION”. See Figure 2 above for further details.
+
+## Transient Failures
+Transient errors arise as a result of a failure beyond the control of the Worker, e.g the database being down or unavailable.
+
+![Transient Failure](images/TransientFailure.png)
+*Figure 3 - Transient failure flow*
+
+1. The Worker will receive a message from the messaging input queue, `worker-input-queue` and will start to process it.
+2. The Worker is unable to process the message (unexpectedly due to some failure outside of its control - e.g. database down).
+3. The failure is reported as transient causing the Worker to be flagged as unhealthy.
+4. The Worker stops accepting work and messages intended for it are queued up within the messaging system. Messages will continue to be queued until the transient issue is addressed.
+5. The Worker continues to retry the message indefinitely.
+6. If the transient error is resolved, the message is acknowledged.
+7. If the transient error is resolved, a `RESULT_SUCCESS` response is placed on the messaging output queue `worker-output-queue` by the worker framework.
+ 
+## InvalidTaskException
+An InvalidTaskException is thrown when the input message is not parsable. This exception type does not cause the Worker to crash. The message is never re-tried. Instead, an `INVALID_TASK` response with error details is placed on the messaging output queue `worker-output-queue` by the worker framework.
+
+## ResultException
+A ResultException can be detected by the Worker as a result of some fault in the Worker logic (e.g. NullPointerException). The Worker will not crash as a result of this kind of error but the original message will never be retried.
```

Comment:
Should the title of this section be `RuntimeException` rather than `ResultException`?

Reply:
Yes it probably should be.